autoload/phpactor/project/project.vim

```diff
@@ -0,0 +1,39 @@
+function! phpactor#project#project#createFromRootPath(primaryRootPath) abort
+  let l:project = {
+        \ 'primaryRootPath': s:normalizeRootPath(a:primaryRootPath),
+        \ 'getPrimaryRootPath': function('s:getPrimaryRootPath'),
+        \ 'containsFile': function('s:containsFile')
+        \ }
+
+  return l:project
+endfunction
+
+function! s:getPrimaryRootPath() dict
+  return l:self.primaryRootPath
+endfunction
+
+function! s:containsFile(filename) dict
+  let l:path = simplify(fnamemodify(a:filename, ':p:h'))
+
+  " @todo Is this working on Windows filesystem?
+  while resolve(l:path) !=# resolve(self.primaryRootPath)
+    let l:path = fnamemodify(l:path, ':h')
+
+    if l:path ==# '/'
+      return v:false
+    endif
+  endwhile
+
+  return v:true
+endfunction
+
+function s:normalizeRootPath(path) abort
+  let l:path = simplify(fnamemodify(a:path, ':p'))
+
+  " @todo better check if it is not an existing directory
+  if empty(glob(l:path.'/'))
```

Comment:
You got `isdirectory()` to help with that ;)

Reply:
I suspected that there is such function but I overlooked it browsing `help file-functions`. 

:+1:

Reply:
If you got `fzf` on your compture and `fzf.vim` I think, you got a great `:Helptags` command to fuzzy find stuff in the doc :)

Reply:
I am used to browse command or buffers history, buffers (using shortcuts) but forget about `:Helptags` ;)